internal/venice-common/src/main/java/com/linkedin/venice/writer/VeniceWriter.java

```diff
@@ -147,9 +148,9 @@ public class VeniceWriter<K, V, U> extends AbstractVeniceWriter<K, V, U> {
   public static final int DEFAULT_MAX_SIZE_FOR_USER_PAYLOAD_PER_MESSAGE_IN_BYTES = 950 * 1024;
 
   /**
-   * The default value of -1 is overwritten with 10 MB, if not specified in the VeniceProperties from the constructor.
+   * The default value of -1 from the store is overwritten with a new default of 10 MB.
    */
-  // public static final int DEFAULT_MAX_RECORD_SIZE_BYTES = 10 * 1024 * 1024;
+  public static final int DEFAULT_MAX_RECORD_SIZE_BYTES = 10 * 1024 * 1024;
```

Comment:
I wonder whether this can be controlled by the backend or not.
For example, we can introduce a controller config: `controller.default.max.record.size.bytes` and Controller can backfill the max record size if the store-level config is absent (-1) in StoreResponse when VenicePushJob is asking for it.
With this way, when we need to change the default max size, we only need to deploy a config change to the Controller and we can control the default max config per cluster.

Reply:
I would be okay with implementing this, but I do find it a bit odd to have a configurable setting that's supposed to be a `"default"`.

Alternatively, I'm hoping we can figure out a good default value that won't need to be changed in the future. We are still waiting on the assembled value size metric to be rolled out to production, so we can see how large the records really are:

- https://github.com/linkedin/venice/pull/1027

If we were to go with this approach, I would prefer this to be done in a follow-up PR to keep this PR focused.

Reply:
Well, this is more like a existing practice.
For example, we do have a config to control the max batch-get limit and by default, it is `-1` in store level, and we control the true default value via a cluster-level config, and in most of the clusters, we set it to be `150`.
We are evaluating whether we can bump the default up to some higher number, and it will be easy to do that as we just need to deploy a config change without changing the code or touching every store.

Reply:
Ah, if it's an existing practice then there are still many more things such as this that I need to familiarize myself with. I'll work on these changes right now, but I feel like creating a separate PR for that and merging that in first? I don't want to overcomplicate this PR.

Ideally it should've been done in the previous PR for just adding the max limit, but I misunderstood in our earlier conversation:
- https://github.com/linkedin/venice/pull/1027

Reply:
Nevermind, I'll just keep it in this PR. I thought the changes would span many files like for the original setting:
- https://github.com/linkedin/venice/pull/1027

But I found [this commit](https://github.com/linkedin/venice/commit/9612ff6f92aa6128384724eecb91f19ea4414b57) and used it as an example, and only a few files were touched.

Reply:
Do we still need this?
When it is `-1`, it indicates that the max record size is infinite, right?

Reply:
Oh I missed `VeniceWriter` when removing stuff from `VenicePushJob`. Changed default value to `-1` now.